lib/rules/_base.js

```diff
@@ -521,16 +522,32 @@ module.exports = class {
       defaults.filePath = this.filePath;
     }
 
-    // perform the node prop expansion only if those properties don't exist in result
+    if (!result.node) {
+      if (!loggedRules.has(this.ruleName)) {
+        let message = `Calling the log method without passing the node is deprecated. Please ensure you pass node in the log method's result.`;
+
+        if (process.env.JEST_WORKER_ID !== undefined) {
+          throw new Error(message);
+        } else {
+          this._console.log(message);
+
+          loggedRules.add(this.ruleName);
+        }
+      }
+    }
+
+    // perform the node property expansion only if those properties don't exist in result
     if (result.node) {
       let node = result.node;
+      delete result.node;
+      // if we've passed node, we want to ensure that we're correctly expanding the properties
+      // to include line, column, and source. If the rule has passed custom values for these,
+      // we want to respect those.
       nodeProps = {
-        line: node.loc && node.loc.start.line,
-        column: node.loc && node.loc.start.column,
-        source: this.sourceForNode(node),
+        line: result.line || (node.loc && node.loc.start.line),
+        column: result.column || (node.loc && node.loc.start.column),
+        source: result.source || this.sourceForNode(node),
```

Comment:
Hmm, I don't think this quite works. `result.line`/`result.column` could be `0` and therefore falsey. Can you add a test case for that?

Reply:
Ya good call. Will fix.